cmd/binlog-collector/collector/collector.go

```diff
@@ -0,0 +1,201 @@
+package collector
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/binlog-collector/db"
+	"github.com/percona/percona-xtradb-cluster-operator/cmd/binlog-collector/storage"
+)
+
+type Collector struct {
+	db              *db.PXC
+	storage         storage.Client
+	lastSet         string // last uploaded binary logs set
+	pxcServiceName  string // k8s service name for PXC, its for get correct host for connection
+	pxcUser         string // user for connection to PXC
+	pxcPass         string // password for connection to PXC
+	lastSetFileName string // name for object where the last binlog set will stored
+	bufferSize      int64  // size of uploading buffer
+}
+
+type Config struct {
+	PXCServiceName string
+	PXCUser        string
+	PXCPass        string
+	S3Endpoint     string
+	S3AccessKeyID  string
+	S3AccessKey    string
+	S3BucketName   string
+	S3Region       string
+	BufferSize     int64
+}
+
+func New(c Config) (*Collector, error) {
+	s3, err := storage.NewS3(c.S3Endpoint, c.S3AccessKeyID, c.S3AccessKey, c.S3BucketName, c.S3Region, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "new storage manager")
+	}
+	lastSetFileName := "last-binlog-set"
+	// get last binlog set stored on S3
+	lastSet, err := s3.GetObject(lastSetFileName)
+	if err != nil {
+		return nil, errors.Wrap(err, "get last set content")
+	}
+
+	return &Collector{
+		storage:         s3,
+		lastSet:         string(lastSet),
+		pxcUser:         c.PXCUser,
+		pxcPass:         c.PXCPass,
+		pxcServiceName:  c.PXCServiceName,
+		lastSetFileName: lastSetFileName,
```

Comment:
this is the static data, it shouldn't be a part of the collector

It should be a package const. Along with the "-gtid-set"

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/622/commits/c2c0f622d3137e742c2823df2fad5c0260e04428